be/src/common/config.h

```diff
@@ -938,6 +938,14 @@ CONF_mInt64(l0_max_mem_usage, "67108864");  // 64MB
 CONF_mInt64(l0_snapshot_size, "16777216"); // 16MB
 CONF_mInt64(max_tmp_l1_num, "10");
 CONF_mBool(enable_parallel_get_and_bf, "true");
+CONF_mBool(enable_pindex_filter, "true");
+CONF_mInt32(max_pages_ratio, "20");
+// use bloom filter in pindex can reduce disk io, but in the following scenarios, we should skip the bloom filter
+// 1. The records to be found are in the index, bloom filter is no usage
+// 2. The records to be found is very small but bloom filter is very large, read bloom filter may cost a lot of disk io
+// So the bloom filter bytes should less than the index data we need to scan in disk, and the default strategy is if bloom
+// filte bytes is less or equal than 10% of pindex bytes, we will use bloom filter to filter some recodes
+CONF_mInt32(max_bf_read_bytes_ratio, "10");
```

Comment:
The bloomfilter and pindex can be estimated accurately. You can have a accurate number to decide whether to use bloomfilter or pindex.

Reply:
> The bloomfilter and pindex can be estimated accurately. You can have a accurate number to decide whether to use bloomfilter or pindex.

We cannot predict the effect of the bloom filter until we have filtered the data. And read bloom filter will add extra disk io if bloom filter has no effect. So in some case(load only one row but historical data amount is very large), reading bloom filter is not a good choice.  So before read bloom filter, i will calculate the bloom filter data size and pindex data size, if bloom filter data size is too large, we skip it.

Reply:
If you can not predict the effect of which index to use, I think in most cases, the user can't get to use which index.
They are not familiar with the implementation of bloomfilter and pkindex.

Reply:
> If you can not predict the effect of which index to use, I think in most cases, the user can't get to use which index. They are not familiar with the implementation of bloomfilter and pkindex.

This configuration does not need to be changed by the user and this configuration is used to make decisions about whether we need to go ahead and filter the data with a bloom filter. 
Actually, this configuration is used to avoid a bad case. For example, the historical data is very large and the bloom filter data size is large too. If we only read one row, the bloom filter data size may be much larger than the real data we need to read. So this configuration does not need to be changed by the user. The purpose of adding this configuration is to ensure we won't get performance degradation after we enable this enhancement.

Reply:
The problem is to load two indices, bloom, and pk index. Could we choose only to use the bloom filter? The primary key is at high cardinality, the bloom filter's effect may be better than the index.

Reply:
> The problem is to load two indices, bloom, and pk index. Could we choose only to use the bloom filter? The primary key is at high cardinality, the bloom filter's effect may be better than the index.

Bloom filter should be better than primary index at most time. But I think we can't  ignore the corner case. If we always enable bloom filter index, we may get performance degradation in real time data load.